### ISSUE TITLE
Communicate sync'd nature of Navigation blocks

### ISSUE THREAD

**Issue by getdave** [WordPress/gutenberg#49158]
The Navigation block has a `ref` attribute which "points" to a Custom Post Type entity which stores the inner blocks for the given menu.

However, unlike Template Parts and Reusable blocks which were given visual affordances in WP 6.2, there is nothing to communicate the "sync'd" nature of the block.

I would like to propose that we add both visual and accessible affordances to indicate the status to better communicate that the items are "saved".

Note that I explored this in https://github.com/WordPress/gutenberg/pull/46412 but it was not accepted. It's worth reviewing however for visual and code options.

Related 

- https://github.com/WordPress/gutenberg/issues/46569
- https://github.com/WordPress/gutenberg/issues/43261

**Comment by jasmussen** [comment]
There's a fair bit of overlap with #43261, and it would be good to find a generic solution one that isn't unique to the navigation block.

I've previously suggested we could make the block appear (in list view, block toolbar and inspector) purple, same as template parts. Also tracked in #47546 and #46569 (seem like unintended duplicates). But it looks like @jameskoster has some [deeper thoughts](https://github.com/WordPress/gutenberg/issues/43261) on when and how to apply this. Jay, do you think the difference between how navigation is synced compared to TPs is significant enough to not make this purple?

For what it's worth, I'm happy to refer to Jay if he feels strongly about that. The fact that the navigation is synced does not feel to me like the standout feature — its intrinsic behavior of loading a menu initially seems to imply this enough that perhaps we really don't need additional indication.

**Comment by jameskoster** [comment]
> Jay, do you think the difference between how navigation is synced compared to TPs is significant enough to not make this purple

I still have some reservations about marking Navigation as synced without some context in the UI. Mostly because it creates inconsistency with the convention that folks are taught in the Template Part / Reusable Block experience, where synced blocks with the same name appear identically in all instances. But in the case of the Navigation block, you could have two blocks in the same document, with the same name, marked as synced, but displaying totally different menus, with different behaviours. 

As things stand, the only thing that's synced in the Navigation block are the menu items in the referenced menu. So unless we're willing to rework Navigation to be more aligned with Template Parts / Reusable Blocks, that's the thing I think it would explore marking as synced. A couple of rough ideas:

Denote the referenced menu in List View / Inspector:

<img width="281" alt="Screenshot 2023-03-17 at 12 25 34" src="https://user-images.githubusercontent.com/846565/225904152-e77aaf9e-30bb-4664-a821-b83a95cba964.png">
<img width="318" alt="Screenshot 2023-03-17 at 12 25 59" src="https://user-images.githubusercontent.com/846565/225904158-63368cec-45a3-4fa2-bd31-feca6193a208.png">

Some educational content in the menu panel, more eloquent than this but you get the idea:

<img width="281" alt="Screenshot 2023-03-17 at 12 30 30" src="https://user-images.githubusercontent.com/846565/225905208-240a3b57-5a7a-4a51-9f0b-d246f2282015.png">

---

All that said, I acknowledge the nuances may be lost on folks, so it's not a super-strong opinion :)

**Comment by jasmussen** [comment]
> Mostly because it creates inconsistency with the convention that folks are taught in the Template Part / Reusable Block experience, where synced blocks with the same name appear identically in all instances. But in the case of the Navigation block, you could have two blocks in the same document, with the same name, marked as synced, but displaying totally different menus, with different behaviours.

Ah, that's a good note. In that light I would lean to keeping things as is, no purple, no chips, but at most a notice like you mocked up, but I'd keep it gray, and maybe shorten the text.

Just to not leave the stone unturned (I'm not sure it's a good idea), but we _could_ theoretically change the name of a navigation block in the list view according to the menu that was loaded. Like TPs and RBs:

<img width="367" alt="Screenshot 2023-03-17 at 13 47 53" src="https://user-images.githubusercontent.com/1204802/225909192-03b8b794-29f1-4fcf-8a4c-4efc6bfa3a5b.png">

**Comment by jasmussen** [comment]
Also good to think about the pattern block here https://github.com/WordPress/gutenberg/issues/48458

**Comment by jameskoster** [comment]
> but we could theoretically change the name of a navigation block in the list view according to the menu that was loaded

I know it's super nuanced, but I think this still fails because the same menu can have entirely different properties, which breaks the 'sync'. 

https://user-images.githubusercontent.com/846565/225913433-ead7f25e-5d7a-46a2-b7af-3a7c09e2e6e1.mp4

The _items_ are synced, but the settings / appearance are not.

**Comment by getdave** [comment]
Great discussion here folks. Thank you.

> The items are synced, but the settings / appearance are not.

Right. This is key. So to be 100% clear this is my understanding:

- Template Parts/Reusable Blocks
    - the block itself is just a "wrapper" - it doesn't have any styling properties.
    - all of the content is sync'd to an entity, including the styling and everything else
- Navigation
    - the block is an actual block (not just a wrapper) and can receive its own styling and other properties on a per instance basis
    - only the items within the block are persisted. 

But there is nuance here, because in actual fact the items _can_ have their own styling (and other properties) which _is_ persisted to the entity. For example, see this hideous Navigation I made which has both custom links and social links which both have styling attached which is saved to the entity. Both Navigation blocks reference the same menu and the styling attributes are shared.



<img width="780" alt="Screenshot 2023-03-17 at 14 31 42" src="https://user-images.githubusercontent.com/444434/225934425-82242b62-8eb6-465f-bdc8-9cb91fc3ab7e.png">

Note however that I applied different styling to the Navigation block itself. It's quite confusing 🤔 

So how do we resolve this?

I think we ought to better communicate which Nav menu is currently in use. Be that Primary, Secondary whatever. 

The next question is do we need to draw a harder line on what is/isn't saved to the menu entity? What is user expectation?

**Comment by talldan** [comment]
Just to note, I do still have this PR open - https://github.com/WordPress/gutenberg/pull/46457 (surprisingly it has no merge conflicts).

It may serve as a good basis for addressing this issue, or could be closed by the outcome.

> But there is nuance here, because in actual fact the items can have their own styling (and other properties) which is persisted to the entity.

Template parts and reusable blocks can also inherit styling from parent blocks like groups, so they don't always appear the same, but I guess they do in isolation. Maybe i'm nitpicking 😄

**Comment by jameskoster** [comment]
> Note however that I applied different styling to the Navigation block itself. It's quite confusing

Yeah, that encapsulates my concern with marking the Navigation block as synced. 

> So how do we resolve this?

Perhaps we wait and see how https://github.com/WordPress/gutenberg/issues/48458 progresses? The designs there might help unify many things, including Navigation, Template Parts, and Reusable Blocks.

**Comment by annezazu** [comment]
As we explore examining this, I wanted to note that I hear feedback from time to time around understanding where menus are used, especially when deleting or making changes. If there's a way we can surface this while we communicate the synced nature that seems ideal!

**Comment by jasmussen** [comment]
I have a feeling that as we surface all synced menus in a list in the site editor > navigation section, it will help here as well, to indicate their synced nature.

**Comment by getdave** [comment]
Circling back to this Issue which has not moved.

My feeling is we need to communicate which Menu is being used on which block. Currently to do that you have to:

- Click on the Navigation block
- Open the block sidebar
- Click on the `List View` tab
- Click on the 3 dots menu
- See the currently selected Menu.

That's not helpful to users.

I believe we should consider addressing this in small steps:

<details>
<summary>Firstly, surface the menu _name_ next to the word `Menu` in the Navigation block's `List View` tab</summary>
![Image](https://github.com/user-attachments/assets/64a3895d-f27f-44cd-a813-3d8f8623afb3)
</details>

<details>
<summary>Secondly, consider surfacing the menu name in the List View next to the Navigation block.</summary>
![Image](https://github.com/user-attachments/assets/5fdd5ded-02c2-407a-982d-bffb755f144a)
</details>



There may also be additional affordances we can offer but I believe this Issue should be renamed and refocused on communicating the name of the associated Menu without overloading the UI for less experienced users who may not need to know about the sync'd nature of Navigation Menus.

**Comment by getdave** [comment]
Whilst I appreciate it's not a top priority, I'd welcome any input on this from @WordPress/gutenberg-design 🙏

**Comment by jasmussen** [comment]
Outside of what's been tried so far, I believe the next steps involve actually improving the navigation block itself. For instance, something like #44486 would make progress on aspects that are intrinsically linked to being synced.

**Comment by getdave** [comment]
I agree. That's why [I'm proposing](https://github.com/WordPress/gutenberg/issues/49158#issuecomment-2460077201) "surface the menu _name_ next to the word `Menu` in the Navigation block's `List View` tab". That doesnt involve redesigning the entire editor UI and is focused on the block itself.

I also agree that https://github.com/WordPress/gutenberg/issues/44486 is important.

**Comment by jasmussen** [comment]
>  "surface the menu name next to the word Menu in the Navigation block's List View tab"

@fcoveram I think you and I briefly talked about this too.

**Comment by fcoveram** [comment]
I just shared [here](https://github.com/WordPress/gutenberg/issues/56245#issuecomment-2488343231) (#56245) the idea I showed you @jasmussen. Nonetheless, putting below the approach to sync pages.

![image](https://private-user-images.githubusercontent.com/895819/388075760-b51197bd-eb83-4c65-81cc-d5c7bd86200d.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIxMDI3ODgsIm5iZiI6MTczMjEwMjQ4OCwicGF0aCI6Ii84OTU4MTkvMzg4MDc1NzYwLWI1MTE5N2JkLWViODMtNGM2NS04MWNjLWQ1YzdiZDg2MjAwZC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjQxMTIwJTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI0MTEyMFQxMTM0NDhaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT0zNTJkOGU4NmE2YmI3OWQ5MzEzN2ZiYTc3OTUwZmIzYzM0OGE3N2ZhODU3NDNkYTAyN2I3ZTkxMDRlZjhlMzQ0JlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCJ9.ZhNFwf2TtEMMio0Q1bU2UZW4MeH4bk-tEDYmLvWQDr8)

**Comment by Mamaduka** [comment]
This also came up during #68446.

Can we start by just marking Navigation as a synced block? Which would enable purple icon coloring and dynamic titles in BlockCard.

**Comment by karmatosed** [comment]
I am going to move this out of the double design label state and just to feedback for now. It has a design mechanism we can use. I think now it's about application, if that is incorrect we can always adjust labels.
